egs2/TEMPLATE/sds1/app.py

```diff
@@ -0,0 +1,797 @@
+
+access_token = os.environ.get("HF_TOKEN")
+ASR_name = None
+LLM_name = None
+TTS_name = None
+ASR_options = []
+LLM_options = []
+TTS_options = []
+Eval_options = []
+upload_to_hub = None
+dialogue_model = None
+
+
+def read_args():
+    global access_token
+    global ASR_name
+    global LLM_name
+    global TTS_name
+    global ASR_options
+    global LLM_options
+    global TTS_options
+    global Eval_options
+    global upload_to_hub
+    global dialogue_model
+    parser = argparse.ArgumentParser(
+        description="Run the app with HF_TOKEN as a command-line argument."
+    )
+    parser.add_argument(
+        "--HF_TOKEN", required=True, help="Provide the Hugging Face token."
+    )
+    parser.add_argument(
+        "--asr_options",
+        required=True,
+        help="Provide the possible ASR options available to user.",
+    )
+    parser.add_argument(
+        "--llm_options",
+        required=True,
+        help="Provide the possible LLM options available to user.",
+    )
+    parser.add_argument(
+        "--tts_options",
+        required=True,
+        help="Provide the possible TTS options available to user.",
+    )
+    parser.add_argument(
+        "--eval_options",
+        required=True,
+        help="Provide the possible automatic evaluation metrics available to user.",
+    )
+    parser.add_argument(
+        "--default_asr_model",
+        required=False,
+        default="pyf98/owsm_ctc_v3.1_1B",
+        help="Provide the default ASR model.",
+    )
+    parser.add_argument(
+        "--default_llm_model",
+        required=False,
+        default="meta-llama/Llama-3.2-1B-Instruct",
+        help="Provide the default ASR model.",
+    )
+    parser.add_argument(
+        "--default_tts_model",
+        required=False,
+        default="kan-bayashi/ljspeech_vits",
+        help="Provide the default ASR model.",
+    )
+    parser.add_argument(
+        "--upload_to_hub",
+        required=False,
+        default=None,
+        help="Hugging Face dataset to upload user data",
+    )
+    args = parser.parse_args()
+    access_token = args.HF_TOKEN
+    ASR_name = args.default_asr_model
+    LLM_name = args.default_llm_model
+    TTS_name = args.default_tts_model
+    ASR_options = args.asr_options.split(",")
+    LLM_options = args.llm_options.split(",")
+    TTS_options = args.tts_options.split(",")
+    Eval_options = args.eval_options.split(",")
+    upload_to_hub = args.upload_to_hub
+    dialogue_model = ESPnetSDSModelInterface(
+        ASR_name, LLM_name, TTS_name, "Cascaded", access_token
+    )
+
+
+read_args()
+api = HfApi()
+nltk.download("averaged_perceptron_tagger_eng")
+
+latency_ASR = 0.0
+latency_LM = 0.0
+latency_TTS = 0.0
+
+text_str = ""
+asr_output_str = ""
+vad_output = None
+audio_output = None
+audio_output1 = None
+LLM_response_arr = []
+total_response_arr = []
+
+
+def handle_eval_selection(
+    option, TTS_audio_output, LLM_Output, ASR_audio_output, ASR_transcript
+):
```

Comment:
I think it is good to add docstrings for each functions. You can easily generate it with chatGPT.

Eg:
```
def handle_eval_selection(self, option, TTS_audio_output, LLM_Output, ASR_audio_output, ASR_transcript):
    """
    Handles the evaluation of a selected metric based on user input and provided outputs.

    Parameters:
    ----------
    option : str
        The evaluation metric selected by the user. Supported options include:
        - "Latency"
        - "TTS Intelligibility"
        - "TTS Speech Quality"
        - "ASR WER"
        - "Text Dialog Metrics"
    TTS_audio_output : np.ndarray
        The audio output generated by the TTS module for evaluation.
    LLM_Output : str
        The text output generated by the LLM module for evaluation.
    ASR_audio_output : np.ndarray
        The audio input/output used for ASR evaluation.
    ASR_transcript : str
        The transcript generated by the ASR module for evaluation.

    Returns:
    -------
    str
        A string representation of the evaluation results. The specific result depends on the selected evaluation metric:
        - "Latency": Latencies of ASR, LLM, and TTS modules.
        - "TTS Intelligibility": A score indicating how intelligible the TTS audio output is.
        - "TTS Speech Quality": A score representing the speech quality of the TTS audio output.
        - "ASR WER": The Word Error Rate (WER) of the ASR output.
        - "Text Dialog Metrics": A combination of perplexity, diversity metrics, and relevance scores for the dialog.

    Raises:
    ------
    ValueError
        If the `option` parameter does not match any supported evaluation metric.

    Example:
    -------
    >>> result = handle_eval_selection(
            option="ASR WER",
            TTS_audio_output=audio_array,
            LLM_Output="Generated response",
            ASR_audio_output=audio_input,
            ASR_transcript="Expected transcript"
        )
    >>> print(result)
    "WER: 12.34%"
    """
```

Reply:
Sounds good! I added docstring to each function!